packages/next/src/client/components/navigation.ts

```diff
@@ -187,7 +187,7 @@ function getSelectedLayoutSegmentPath(
   if (!node) return segmentPath
   const segment = node[0]
   const segmentValue = Array.isArray(segment) ? segment[1] : segment
-  if (!segmentValue || segmentValue === '__PAGE__') return segmentPath
+  if (!segmentValue || segmentValue.startsWith('__PAGE__')) return segmentPath
```

Comment:
Doesn't this just reverse what @timneutkens did [here](https://github.com/vercel/next.js/pull/47492/files#diff-77f06665ebc0df2289daa6fb683f5d2c160bbfdf90a9f3ba9cf6b02bd0b7d57aR152)?